app/main.py

```diff
@@ -1,3 +1,133 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+import math
+import datetime
+
+from app.customer import Customer
+from app.shop import Shop
+
+
+def distance_to_shop(costumer: Customer, shop: list) -> float:
+    costumer_x, costumer_y = costumer.location
+    shop_x, shop_y = shop
+    result = math.sqrt((costumer_x - shop_x) ** 2 + (costumer_y - shop_y) ** 2)
+    return result
+
+
+def consumption_for_the_ride(
+        distance: float,
+        fuel_consumption: float,
+        fuel_price: float
+) -> float:
+    result = fuel_consumption * (distance * 2) / 100
+    return result * fuel_price
+
+
+def consumption_for_purchase(
+        costumer: Customer,
+        shop: Shop,
+        fuel_price: float
+) -> float:
+    milk = costumer.milk() * shop.milk_price()
+    bread = costumer.bread() * shop.bread_price()
+    butter = costumer.butter() * shop.butter_price()
+    distance = distance_to_shop(costumer, shop.location)
+    ride_cost = consumption_for_the_ride(
+        distance, costumer.car_fuel_consumption(), fuel_price
+    )
+    sum_purchase = milk + bread + butter + ride_cost
+    return round(sum_purchase, 2)
+
+
+def shop_trip() -> None:
+    list_of_costumers_instance = []
```

Comment:
list_of_costumers_instance rename to customers
No need to add list_ and _instance it is redundant

Reply:
The same with shops